docker_cli/_utils.py

```diff
@@ -0,0 +1,301 @@
+import io
+import random
+import re
+from contextlib import contextmanager
+from shlex import split
+from string import ascii_lowercase, digits
+from subprocess import DEVNULL, CalledProcessError, run
+from threading import Lock
+from typing import Generator, Optional, Tuple
+
+from ._image import Image
+from ._package_manager import (
+    PackageManager,
+    get_package_manager,
+    get_supported_package_managers,
+)
+from ._url_reader import URLReader, get_supported_url_readers, get_url_reader
+
+
+def universal_tag_prefix() -> str:
+    """
+    Returns a prefix for tags generated by the system.
+
+    Returns
+    -------
+    str
+        The prefix.
+    """
+    return "dcli"
+
+
+@contextmanager
+def temp_image(
+    base: str,
+    stdout: Optional[io.TextIOBase] = None,
+    stderr: Optional[io.TextIOBase] = None,
+) -> Generator[Image, None, None]:
```

Comment:
Looks like this just yields an `Image` object. I don't think I understand the `None, None` part here

```suggestion
) -> Generator[Image]:
```

Reply:
This is a quirk of `typing.Generator` - mypy will complain if I don't put 3 type arguments in.

The reason they give (which I don't really like) is that it wants a yield type, a return type, and a send type. This one is only used with yielding, so it only has the initial of the three types.

The only other way I've seen of doing this that doesn't make mypy angry is `Iterable[Image]` - it seems like a bit of a misleading thing to type as, but it would also work if you prefer.

Reply:
Thanks for the explanation. I should've read the docs for `typing.Generator` before commenting.

I'd be equally happy with keeping it as is or changing to `Iterator[Image]`.